cli/src/merge_tools/builtin.rs

```diff
@@ -56,6 +56,21 @@ enum FileContents {
     },
 }
 
+impl FileContents {
+    fn is_empty(&self) -> bool {
+        match self {
+            // Being absent is different from being present and empty.
+            FileContents::Absent => false,
+            FileContents::Text {
+                contents: _,
+                hash: _,
+                num_bytes,
+            } => *num_bytes == 0,
+            FileContents::Binary { hash: _, num_bytes } => *num_bytes == 0,
+        }
+    }
```

Comment:
Maybe this can be inlined to caller? It's unclear why absent file isn't empty.

If I understand it, the condition to render the mode section is one side is `Absent` and the other side is `Text { num_bytes: 0 }`. If there were an empty binary file, I think the entry would be selectable.

Oh, btw, the binary section could be abused to render absent<->empty diff, but I'm not sure if it works and if it provides better UX.

Reply:
Good point about an empty binary file being selectable. I haven't tried it but I suspect you're right. I used the mode section because arxanas [mentioned on the bug](https://github.com/martinvonz/jj/issues/3016#issuecomment-1937857687) that new files should show a mode change. Rendering them as 0-byte binary files could work as well though.

It's not totally clear what `Absent` means, but it seems that it should mean that the file doesn't exist at all, otherwise it's hard to reason about whether a file exists or not after the user selects the changes they want.

Reply:
If we keep `is_empty()` function, I slight prefer making it return `Option<bool>`, and match `(None, Some(true))` pair.

I totally agree with the `Absent` definition.

Reply:
Ok I changed the return value to `Option<bool>` and moved the function to the `FileInfo` struct. I also added a `debug_assert!` to ensure that if the mode is absent the contents are also absent.

Reply:
> I used the mode section because arxanas mentioned on the bug that new files should show a mode change. Rendering them as 0-byte binary files could work as well though.

comment(non-blocking): I think using either is justified. I was thinking of the mode change because I think that's how Git renders newly-added files with no contents (well, it probably *always* renders modes in diffs, but it's usually not notable except when there are other changes as well).